File: DoItTest/ClientApp/src/domain/tests/items/anserOption/radioButtonAnserOption.ts
import { AnserOption } from "./anserOption"

export class RadioButtonAnserOption extends AnserOption {
    constructor(
        public id: string,
        public testItemId: string,
        public title: string,
        public isTrue: boolean
    ) {
        super(id, testItemId)
    }
}

export function mapRadioButtonAnserOption(value: any): RadioButtonAnserOption {
    return new RadioButtonAnserOption(value.id, value.testItemId, value.title, value.isTrue)
}

export function mapRadioButtonAnserOptions(value: any[]): RadioButtonAnserOption[] {
    return value.map(mapRadioButtonAnserOption)
}

File: DoItTest.Domain/Tests/TestItems/AnswerOptions/RadioButtonAnserOption.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DoItTest.Domain.Tests.TestItems.AnswerOptions
{
    public class RadioButtonAnserOption : AnswerOption
    {
        public String Title { get; }
        public Boolean IsTrue { get; }

        public RadioButtonAnserOption(Guid id, Guid testItemId, String title, Boolean isTrue) : base(id, testItemId)
        {
            Title = title;
            IsTrue = isTrue;
        }
    }
}
